scenariorun/src/main/kotlin/com/cosmotech/scenariorun/ContainerFactory.kt

```diff
@@ -1248,8 +1250,10 @@ internal class ContainerFactory(
         }
       }
 
-  private fun getImageName(registry: String, repository: String, version: String? = null): String {
-    val repoVersion = if (version == null) repository else "${repository}:${version}"
+  private fun getImageName(registry: String, repository: String?, version: String? = null): String {
+    if (repository == null) throw IllegalStateException("Solution repository is not defined")
+    val repo = repository ?: "ERROR"
```

Comment:
if `repository` is null, you throws an error, `repo` cannot have the value `ERROR`

Reply:
It must be set for Kotlin in order to handle concurrent access but I may find a way to have a better code without this var set.